src/GraphQL/Execution/Errors/ExecutionError.cs

```diff
@@ -107,5 +120,15 @@ public static TError AddLocation<TError>(this TError error, ASTNode? abstractNod
             error.AddLocation(Location.FromLinearPosition(document.Source, abstractNode.Location.Start));
             return error;
         }
+
+        /// <summary>
+        /// Adds an extension to <see cref="ExecutionError.Extensions"/>.
+        /// </summary>
+        public static TError WithExtension<TError>(this TError error, string key, object? value)
```

Comment:
I don't like the fact that now an error is going to have a `.AddExtension` and a `.WithExtension` method (essentially).  How can we fix this?  Make `AddExtension` internal or move the code into this method?  Remove this extension method?

Reply:
I expected such comment. See `AddLocation` instance method and extension method. They have different signatures so there are no problems. I would like to have single instance method but then I could not chain calls. I removed instance method and rename extension method.

Reply:
I mean could not chain calls **without loosing initial type of error**.